transport/ice/IceSession.h

```diff
@@ -239,31 +250,36 @@ class IceSession
     class CandidatePair
     {
     public:
-        CandidatePair(const IceConfig& config,
+        CandidatePair(IceSession* iceSession,
             const EndpointInfo& localEndpoint,
             const IceCandidate& local,
             const IceCandidate& remote,
-            StunTransactionIdGenerator& idGenerator,
-            const SessionCredentials& credentials,
-            crypto::HMAC& hmacComputerRemote,
-            const std::string& name,
             bool gathering);
 
         CandidatePair& operator=(const CandidatePair&) = delete;
 
         int64_t nextTimeout(uint64_t now) const;
         void processTimeout(uint64_t now);
 
-        void restartProbe(const uint64_t now);
+        void restartProbe(const uint64_t now, bool releaseNomination);
+        int64_t nominatedInProgressNextTimedout(const uint64_t now) const;
         void send(uint64_t now);
         uint64_t getPriority(IceRole role) const;
         bool isFinished() const { return state == ProbeState::Succeeded || state == ProbeState::Failed; }
+        bool isReadable(uint64_t now) const
```

Comment:
A better name is perhaps "isViable".
Meaning: "Capable of success or continuing effectiveness"